linker-interface/shared/src/test/scala/org/scalajs/linker/interface/ReportToLinkerOutputAdapterTest.scala

```diff
@@ -179,6 +179,40 @@ class ReportToLinkerOutputAdapterTest {
           writeOut.content("sm"))
     }
   }
+
+  @Test // #4271
+  def testNoPublicModules(): AsyncResult = await {
+    val writeOut = new WriteOnlyOutputDirectory()
+
+    val emptyReport = new ReportImpl(Nil)
+
+    val legacyOutput = LinkerOutput(new OutputFileImpl("js", writeOut))
+      .withSourceMap(new OutputFileImpl("sm", writeOut))
+      .withSourceMapURI(new URI("http://example.org/my-source-map-uri"))
+      .withJSFileURI(new URI("http://example.org/my-js-file-uri"))
```

Comment:
I think there should also be a test where we don't ask for a source map, since it's a different code path in the implementation.

Otherwise LGTM